### ISSUE TITLE
Subresource operations security voter on current object

### ISSUE THREAD

**Issue by rsereir** [api-platform/api-platform#1944]
**API Platform version(s) affected**: x.y.z

**Description**  
When i add security settings of subresource operation with an object parameter, i get "Paginator" object but not my current entity

**How to reproduce**  
add this settings on my subresource:
```
subresourceOperations={
 *         "api_articles_inventories_get_subresource"={
 *             "normalization_context"={"groups"={"edm:article-inventories:list"}},
 *             "security"="is_granted('MARKET_CATALOG_VIEW', object)",
 *         }
 *     }
```

**Comment by wcoppens** [comment]
Yep, seem to have the same issue with Graphql and oneToMany relations. 
In our case we have:

User --- OneToMany ---> BuildingUserMap <-- OneToMany --- Building.

The relationshsop from User to BuildingUserMap, triggers a graphql `item_query` security setting with a `Paginator` instance. Interestingly the building to BuildingUserMap does correctly trigger the `collection_query` security setting of the BuildingUserMap. Both relationships vice versa are nearly identical, just can't seem to find the solution yet.

**Comment by soyuka** [comment]
Closing as stale per #2987 (no activity since 2021-10-06). This repository is transitioning to a CLI installer. If this still reproduces against a current release of API Platform, please open a new issue on `api-platform/core` with a minimal reproduction.
